library/std/src/io/error.rs

```diff
@@ -565,6 +565,12 @@ impl Error {
     /// other standard library functions may call platform functions that may
     /// (or may not) reset the error value even if they succeed.
     ///
+    /// If this is used in a case where no error has yet occurred in a program,
+    /// e.g. right after the beginning of `fn main`,
+    /// then in principle any possible Error may be returned.
+    /// The error code may have been set by a previous program (e.g. `execve`)
+    /// or the OS may have initialized it to an arbitrary, even random, value.
```

Comment:
Thats not how `main` works; its not actually your program's entry point. That is actually located [here](https://github.com/rust-lang/rust/blob/master/library/std/src/rt.rs#L159) in the standard library and it already does some syscalls (on unix for example [this `init` function](https://github.com/rust-lang/rust/blob/master/library/std/src/sys/unix/mod.rs#L53) gets run)

Reply:
Hm, this is kind of not guaranteed (a platform may have an empty `init` function) but the most important point is that even if the std calls OS functions pre user main, there's no guarantee that these functions set the last OS error code.

That said, I'm unsure if this paragraph adds anything that isn't already stated in the previous one: 

> This should be called immediately after a call to a platform function, otherwise the state of the error value is indeterminate

Surely this covers this case adequately without getting into the weeds of what any particular platform may or may not do?